src/Test/All.hs

```diff
@@ -52,14 +52,16 @@ test CmdTest{..} main dataDir files = do
         wrap "Hint names" $ mapM_ (\x -> do progress; testNames $ snd x) testFiles
         wrap "Hint annotations" $ forM_ testFiles $ \(file,h) -> do progress; testAnnotations h file
         when cmdTypeCheck $ wrap "Hint typechecking" $
-            progress >> testTypeCheck cmdDataDir cmdTempDir [h | (file, h) <- testFiles, takeFileName file /= "Test.hs"]
+            progress >> testTypeCheck cmdDataDir cmdTempDir (hs testFiles)
         when cmdQuickCheck $ wrap "Hint QuickChecking" $
-            progress >> testQuickCheck cmdDataDir cmdTempDir [h | (file, h) <- testFiles, takeFileName file /= "Test.hs"]
+            progress >> testQuickCheck cmdDataDir cmdTempDir (hs testFiles)
 
         when (null files && not hasSrc) $ liftIO $ putStrLn "Warning, couldn't find source code, so non-hint tests skipped"
         getIdeas
     whenLoud $ mapM_ print ideas
     return failures
+    where
+      hs testFiles = [h | (file, h) <- testFiles, takeFileName file /= "Test.hs"]
```

Comment:
Did this get caught by the duplicate check? If so, did you make the duplicate check more powerful? Rather than generalise this why, why not `let hs = ...` before `cmdTypeCheck` and then use it twice? That avoids duplicate compilation as well as duplicate code.

Reply:
It got caught by the duplicate check. Evidently, yes, it got more powerful (not by design). I can change particular phrase along these lines this if you think it important.

Reply:
I think this instance is worth changing. Interesting to know it got a bit more powerful, but as long as the tests still pass, this instance is not an unreasonable thing to say has too much duplication.

Reply:
Fixed